core/src/test/scala/cromwell/util/JsonEditorSpec.scala

```diff
@@ -2,96 +2,245 @@ package cromwell.util
+    val expectedMetadata =
+      """
+        |{
+        |  "workflowName": "main_workflow",
+        |  "actualWorkflowLanguageVersion": "draft-2",
+        |  "submittedFiles": {
+        |    "workflow": "import \"sub_workflow_hello_world_import.wdl\" as sub\n\nworkflow main_workflow {\n    call sub.wf_hello { input: wf_hello_input = \"sub world\" }\n    output {\n        String main_output = wf_hello.salutation\n    }\n}",
+        |    "root": "",
+        |    "options": "{\n\n}",
+        |    "inputs": "{}",
+        |    "labels": "{}",
+        |    "imports": {
+        |    }
+        |  },
+        |  "outputs": {
+        |    "main_workflow.main_output": "Hello sub world!"
+        |  },
+        |  "workflowRoot": "/home/dan/cromwell/cromwell-executions/main_workflow/757d0bcc-b636-4658-99d4-9b4b3767f1d1",
+        |  "id": "757d0bcc-b636-4658-99d4-9b4b3767f1d1",
+        |  "inputs": {},
+        |  "labels": {
+        |    "cromwell-workflow-id": "cromwell-757d0bcc-b636-4658-99d4-9b4b3767f1d1"
+        |  },
+        |  "submission": "2019-07-22T13:32:02.123-04:00",
+        |  "status": "Succeeded",
+        |  "end": "2019-07-22T13:32:41.529-04:00",
+        |  "start": "2019-07-22T13:32:20.434-04:00"
+        |}
+        |
+        |""".stripMargin
+    val expected = parseString(expectedMetadata)
+    actual shouldEqual expected
+  }
 
-  "Json Munger" should "remove excludes" in {
-      val either = testJsonAndGetKeys(includeExcludeJson(_, None, Some(NonEmptyList.one("foo"))))
-      assert(either.right.get.head === "other")
-    }
+  it should "remove nested excludes in workflows, calls and subworkflows" in {
+    // jq -M 'del(..|.callCaching? //empty | .hashes["runtime attribute"]["docker"]) | del(..|.executionStatus? //empty)' gratuitous_subworkflows.json > nested_excludes_gratuitous_subworkflows.json
+    // That jq one-liner knows nothing about workflows, calls or subworkflows; it just happens to get the right answer in this case.
+    val actual = excludeJson(gratuitousSubworkflowJson, NonEmptyList.of("callCaching:hashes:runtime attribute:docker", "executionStatus")).get
+    val expected = parseMetadata("nested_excludes_gratuitous_subworkflows.json")
 
-  it should "remove nested keys excludes" in {
-    val either = testJson(excludeJson(_, NonEmptyList.one("deep")))
-    assert(either.right.get.hcursor.downField("nested").downField("inner").keys.get.head === "keepme")
+    actual shouldEqual expected
   }
 
-  it should "remove multiple nested keys excludes" in {
-    val either = testJson(excludeJson(_, NonEmptyList.of("deep", "wildcard")))
-    val keys = either.right.get.hcursor.downField("nested").downField("inner").keys.get
-    assert(keys.head === "keepme")
-    assert(keys.size === 1)
+  it should "remove excludes from jobs and workflows" in {
+    val actual = excludeJson(helloWorldJson, NonEmptyList.of("outputs", "executionEvents")).get
```

Comment:
Maybe not in this particular test, but maybe we could somewhere test that `excludeKey=wf_hello_input` does **not** exclude the workflow output that happens to be called `wf_hello_input` (because it wasn't qualified properly)?

Reply:
yes this and many other things should be tested.